### PR TITLE
Adds Urban Dictionary Plugin

### PR DESCRIPTION
**Pull request by caleywoods**
This adds an Urban Dictionary plugin so users can get definitions from
the Urban Dictionary site via the /ud command to the bot.

### CODE REVIEW

**Comment by caleywoods** [comment]
This pull request is for #126 for Hacktoberfest.

**Comment by CapacitorSet** [comment]
The CI failure is ok, I had forgotten to update the test suite and it didn't go into your fork.

Thanks for your contribution! It seems alright, but I'll give it a try this GMT-afternoon and merge it.

File: src/plugins/UrbanDictionary.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+const Plugin = require('../Plugin');
+const request = require('request-promise');
+
+module.exports = class UrbanDictionary extends Plugin {
+    static get plugin() {
+        return {
+            name: "UrbanDictionary",
+            description: "Fetches the Urban Dictionary definition for a supplied word or phrase.",
+            help: ` commands:
```
Comment (CapacitorSet) [inline src/plugins/UrbanDictionary.js]: For consistency, I think it's better if you only include the list of commands, without the "commands:" header.

File: src/plugins/UrbanDictionary.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+                        if (data.result_type === 'no_results') {
+                            msg = `Sorry, I was unable to find results for: ${args.join(' ')}.`;
+                        } else {
+                            msg =
+                                `*${def.word}*: ${def.definition}\n\n_${def.example}_`;
+                        }
+
+                        this.sendMessage(message.chat.id, msg, opts);
+                    });
```
Comment (CapacitorSet) [inline src/plugins/UrbanDictionary.js]: It lacks a .catch statement: if the request fails for whatever reason, displaying a short error message (possibly with an error on the command line, this.log.error/this.log.warn) is more user-friendly than not showing anything.

**Review by CapacitorSet — CHANGES_REQUESTED** [review]

**Comment by caleywoods** [comment]
Thanks for "catching" 😄 my missing catch statement. I guess I picked a bad plugin to base my help on in `Quote.js`.

You asked me to use the Wiki and that seemed complete enough for a simple plugin like this. Between the wiki and looking at a few of the existing plugins, as well as the `node-telegram-bot-api` library in `node_modules` to find an example of Promise based requests, was enough to get this working.

File: src/plugins/UrbanDictionary.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+const Plugin = require('../Plugin');
+const request = require('request-promise');
+
+module.exports = class UrbanDictionary extends Plugin {
+    static get plugin() {
+        return {
+            name: "UrbanDictionary",
+            description: "Fetches the Urban Dictionary definition for a supplied word or phrase.",
+            help: `/ud <query> returns the definition for a given word or phrase`
```
Comment (crisbal) [inline src/plugins/UrbanDictionary.js]: What about using 

```
"`/ud <query>` returns the definition for a given word or phrase"`
```

?

**Comment by crisbal** [comment]
Looks good to me. I just added a minor suggestion on the help message (so the command will be formatted in the help). If you want you can fix it and I (or @CapacitorSet) will merge.

**Comment by caleywoods** [comment]
Added your suggestion to the verbiage. Thanks for looking out.

**Pull request merged**